images/management/commands/import-tandemvault-tags.py

```diff
@@ -0,0 +1,157 @@
+    tags_updated = 0
+    tags_deleted = 0
+
+    def add_arguments(self, parser):
+        parser.add_argument(
+            '--file',
+            type=file,
+            help='CSV file containing existing tag and synonym information from Tandem Vault',
+            dest='file',
+            required=True
+        ),
+        parser.add_argument(
+            '--delete-stale',
+            type=bool,
+            help='Whether or not stale tags (tags not assigned to any Images) should be deleted.',
+            dest='delete-stale',
+            default=False,
+            required=False
+        )
+
+    def handle(self, *args, **options):
+        self.tandemvault_tags_csv = options['file']
+        self.do_delete_stale = options['delete-stale']
+
+        # Start a timer for the bulk of the script
+        self.exec_time = 0
+        self.exec_start = timeit.default_timer()
+
+        # Process the CSV
+        self.load_tandemvault_tags()
+        self.process_tandemvault_tags()
+
+        # Delete stale Tandem Vault image tags
+        if self.do_delete_stale:
+            self.delete_stale()
+
+        # Stop timer
+        self.exec_end = timeit.default_timer()
+        self.exec_time = datetime.timedelta(seconds=self.exec_end - self.exec_start)
+
+        # Print the results
+        self.progress_bar.finish()
+        self.print_stats()
+
+        return
+
+    '''
+    Load a CSV of Tandem Vault tags + synonyms.
+    '''
+    def load_tandemvault_tags(self):
+        try:
+            csv_reader = csv.DictReader(self.tandemvault_tags_csv)
+        except csv.Error:
+            logging.error(
+                '\n ERROR reading CSV: CSV does not have valid headers or is malformed.')
+            return
+
+        for row in csv_reader:
+            self.tags.append(row)
+
+        self.tag_count = len(self.tags)
+
+    '''
+    Process loaded Tandem Vault tags.
+    '''
+    def process_tandemvault_tags(self):
+        for tandemvault_tag in self.tags:
+            name = self.clean_tag_name(tandemvault_tag['name'])
+            # Empty tags can exist in this data for whatever reason;
+            # skip them if present
+            if not name:
+                continue
+
+            synonyms = filter(None, [self.clean_tag_name(synonym) for synonym in json.loads(tandemvault_tag['synonym_names'])])
+
+            tag, created = ImageTag.objects.get_or_create(name=name, source=self.source)
+            if created:
+                self.tags_created += 1
+            else:
+                self.tags_updated += 1
+
+            # Remove existing Tandem Vault synonym relationships
+            existing_synonyms = tag.synonyms.filter(source=self.source)
+            if existing_synonyms:
+                tag.synonyms.remove(*existing_synonyms)
+
+            # Assign fresh Tandem Vault synonym relationships
+            for synonym in synonyms:
+                s, created = ImageTag.objects.get_or_create(name=synonym, source=self.source)
```

Comment:
Should we track and print out in the stats the number of synonyms added to give some insight to this portion of the importer?

Reply:
Added.